src/object-store/src/layers/lru_cache/read_cache.rs

```diff
@@ -275,7 +262,7 @@ impl<I: Access> Read for ReadCacheReader<I> {
                     }
                     Err(_) => {
                         OBJECT_STORE_LRU_CACHE_MISS.inc();
-                        self.inner_reader.read_at(offset, limit).await
+                        self.inner_reader.read().await
```

Comment:
I guess we need `read` rather than `read_all` here. Because we are in the implementation of `read`. cc @Xuanwo please confirm.